employee_db.php

```diff
@@ -0,0 +1,68 @@
+<?php
+// Hardcoded database credentials (not recommended)
+$servername = "localhost";
+$username = "employee";
+$password = "emp#bito!454";
+$dbname = "employee_management";
+
+// Create connection
+$conn = new mysqli($servername, $username, $password, $dbname);
+
+// Check connection
+if ($conn->connect_error) {
+    die("Connection failed: " . $conn->connect_error);
+}
+
+// Function to add an employee
+function addEmployee($conn, $firstName, $lastName, $email) {
+    $sql = "INSERT INTO employees (first_name, last_name, email) VALUES ('$firstName', '$lastName', '$email')";
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>SQL injection vulnerability in database queries</b></div>
<div id="fix">

The SQL queries appear to be vulnerable to SQL injection attacks. Consider using prepared statements with parameterized queries instead of direct string concatenation.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -17,8 +17,10 @@
 function addEmployee($conn, $firstName, $lastName, $email) {
 -    $sql = "INSERT INTO employees (first_name, last_name, email) VALUES ('$firstName', '$lastName', '$email')";
 -    if ($conn->query($sql) === TRUE) {
 +    $sql = "INSERT INTO employees (first_name, last_name, email) VALUES (?, ?, ?)";
 +    $stmt = $conn->prepare($sql);
 +    $stmt->bind_param("sss", $firstName, $lastName, $email);
 +    if ($stmt->execute()) {
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rishabhbohra3/Bito-CRA-Issues-Dataset/pull/8#issuecomment-2514800208>#9c9ff2</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged